woocommerce-services.php

```diff
@@ -827,31 +828,68 @@ public function load_admin_dependencies() {
 			require_once __DIR__ . '/classes/class-wc-connect-debug-tools.php';
 			new WC_Connect_Debug_Tools( $this->api_client );
 
-			require_once __DIR__ . '/classes/class-wc-connect-settings-pages.php';
-			$settings_pages = new WC_Connect_Settings_Pages( $this->api_client, $this->get_service_schemas_store() );
-			$this->set_settings_pages( $settings_pages );
-
 			$schema   = $this->get_service_schemas_store();
 			$settings = $this->get_service_settings_store();
 			$logger   = $this->get_logger();
 			$this->set_help_view( new WC_Connect_Help_View( $schema, $this->taxjar, $settings, $logger ) );
 			add_action( 'admin_notices', array( WC_Connect_Error_Notice::instance(), 'render_notice' ) );
 			add_action( 'admin_notices', array( $this, 'render_schema_notices' ) );
 
-			// Add WC Admin Notices.
-			if ( self::can_add_wc_admin_notice() ) {
-				require_once __DIR__ . '/classes/class-wc-connect-note-dhl-live-rates-available.php';
-				WC_Connect_Note_DHL_Live_Rates_Available::init( $schema );
+			if ( ! $this->is_wc_shipping_activated() ) {
+				// We only use the settings page for shipping since tax settings are part of
+				// the core "WooCommerce > Settings > Tax" tab.
+				require_once __DIR__ . '/classes/class-wc-connect-settings-pages.php';
+				$settings_pages = new WC_Connect_Settings_Pages( $this->api_client, $this->get_service_schemas_store() );
+				$this->set_settings_pages( $settings_pages );
+
+				// Add WC Admin Notices.
+				if ( self::can_add_wc_admin_notice() ) {
+					require_once __DIR__ . '/classes/class-wc-connect-note-dhl-live-rates-available.php';
+					WC_Connect_Note_DHL_Live_Rates_Available::init( $schema );
+				}
 			}
 		}
 
 		/**
 		 * Hook plugin classes into WP/WC core.
 		 */
 		public function attach_hooks() {
+			add_action( 'rest_api_init', array( $this, 'rest_api_init' ) );
+			add_action( 'rest_api_init', array( $this, 'wc_api_dev_init' ), 9999 );
+
+			add_action( 'admin_enqueue_scripts', array( $this->nux, 'show_pointers' ) );
+			add_filter( 'plugin_action_links_' . plugin_basename( __FILE__ ), array( $this, 'add_plugin_action_links' ) );
+			add_action( 'enqueue_wc_connect_script', array( $this, 'enqueue_wc_connect_script' ), 10, 2 );
+			add_filter( 'woocommerce_admin_reports', array( $this, 'reports_tabs' ) );
+
+			$tracks = $this->get_tracks();
+			$tracks->init();
+
+			$this->taxjar->init();
+			$this->paypal_ec->init();
+
+			// Primary condition to initiate shipping.
+			if ( ! $this->is_wc_shipping_activated() ) {
+				$this->init_shipping();
```

Comment:
❔ – Curiosity

It is hard to determine what code should go into this function. But the code in this `attach_hooks()` function makes sense, all shipping-related things are now inside `init_shipping()` 👍